runtime/mixed.inl

```diff
@@ -44,22 +57,39 @@ void mixed::copy_from(mixed &&other) {
 
 template<typename T>
 void mixed::init_from(T &&v) {
-  auto type_and_value_ref = get_type_and_value_ptr(v);
-  type_ = type_and_value_ref.first;
-  auto *value_ptr = type_and_value_ref.second;
-  using ValueType = std::decay_t<decltype(*value_ptr)>;
-  new(value_ptr) ValueType(std::forward<T>(v));
+  if constexpr (inner_type_of_class_instance<std::decay_t<T>>::has) {
+    may_be_mixed_base * ptr = dynamic_cast<may_be_mixed_base*>(v.get());
+    if (unlikely(!ptr)) {
+      php_error("Internal compiler error. Trying to set invalid object to mixed");
```

Comment:
It's not a compiler error

Reply:
Logically, it's compiler error. Compiler shouldn't build such a code, so it's kind of assert in runtime. 
Invariant is like "If compiler code written correctly then this error should never happen".

Reply:
It's either a compiler error that happens at compile-time, or runtime error that happens in runtime. It may mislead a programmer that their program crashes in runtime with `compiler error` message.